sw/device/silicon_creator/lib/epmp.c

```diff
@@ -0,0 +1,60 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/silicon_creator/lib/epmp.h"
+
+#include "sw/device/lib/base/csr.h"
+
+/**
+ * Extern declarations of inline functions.
+ */
+void epmp_state_configure_tor(epmp_state_t *state, uint32_t entry,
+                              epmp_region_t region, epmp_perm_t perm);
+void epmp_state_configure_na4(epmp_state_t *state, uint32_t entry,
+                              epmp_region_t region, epmp_perm_t perm);
+void epmp_state_configure_napot(epmp_state_t *state, uint32_t entry,
+                                epmp_region_t region, epmp_perm_t perm);
+
+bool epmp_state_check(const epmp_state_t *s) {
+  bool result = true;
+#define CHECK_CSR(reg, value) \
+  do {                        \
+    uint32_t csr;             \
+    CSR_READ(reg, &csr);      \
+    result &= csr == (value); \
```

Comment:
I take it you don't want to early return here?

Reply:
I don't think it matters either way yet. We'll need to revisit this as we harden this code though. I've left it as-is for now.